lib/ansible/modules/windows/win_chocolatey.ps1

```diff
@@ -271,23 +271,21 @@ Function Install-Chocolatey {
         # locate the newly installed choco.exe
         $choco_app = Get-Command -Name choco.exe -CommandType Application -ErrorAction SilentlyContinue
         if ($null -eq $choco_app) {
-            $choco_path = $env:ChocolateyInstall
-            if ($null -ne $choco_path) {
-                $choco_path = "$choco_path\bin\choco.exe"
-            } else {
-                $choco_path = "$env:SYSTEMDRIVE\ProgramData\Chocolatey\bin\choco.exe"
+            $choco_dir = $env:ChocolateyInstall
+            if ($null -eq $choco_path) {
+                $choco_dir = "$env:SYSTEMDRIVE\ProgramData\Chocolatey"
             }
-
-            $choco_app = Get-Command -Name $choco_path -CommandType Application -ErrorAction SilentlyContinue
+            $choco_app = Get-Command -Name "$choco_dir\bin\choco.exe" -CommandType Application -ErrorAction SilentlyContinue
         }
     }
+
     if ($module.CheckMode -and $null -eq $choco_app) {
         $module.Result.skipped = $true
         $module.Result.msg = "Skipped check mode run on win_chocolatey as choco.exe cannot be found on the system"
         $module.ExitJson()
     }
 
-    if (-not (Test-Path -Path $choco_app.Path)) {
+    if ($null -eq $choco_app -or -not (Test-Path -LiteralPath $choco_app.Path)) {
```

Comment:
This first part might be the actual fix and the changes you've made above really aren't related to the problem.

Reply:
Correct, but they made me understand that we did not need $choco_path at all. So not only did I fix the issue, I also improved the readability of the code which deceived me. ANd it deceived you as well, because you also commented at first that we had to use $choco_path further on.